### PR TITLE
05.System-updates-Yocto-Project/05.Customize-Mender/: mender-shell

### PR DESCRIPTION
**Pull request by mchalski**
I put it together based on advice I gathered from client team members and existing docs. please take a good look as I'm far from an expert.

depends on https://github.com/mendersoftware/mender-docs/pull/1242 (for a link to mender-shell config options).

https://tracker.mender.io/browse/MEN-4098

---
# External Contributor Checklist

🚨 Please review the [guidelines for contributing](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md) to this repository.

- [ ] Make sure that all commits are signed with [`git --signoff`](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work). Also note that the signoff author must match the author of the commit.

### Description

Please describe your pull request.

Thank you!

### CODE REVIEW

File: 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md [hunk 1]
```diff
@@ -204,3 +204,35 @@ for more information.
 ## Delta update support
 
 Delta update support is covered in [its own sub section](01.Delta-update-support/docs.md).
+
+
+## Addons 
+
+### Mender-shell
+
+Mender-shell is not enabled by default, unless you are building with the demo layer.
```
Comment (merlin-northern) [inline 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md:213]: passive. suggested change: "Mender-shell is not enabled by default" -> "_We do not enable mender-shell_ by default". and I reallt have no idea wether we use Mender-shell or mender-shell and shall we treat it as a command, in that case "the mender-shell" or the name of the addon, in that case we can drop the article.

**Review by merlin-northern — COMMENTED** [review]
I will leave the yocto related things for the experts to review.

File: 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md (hunk 1 above)
Comment (mchalski) [reply 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md]: fixed

File: 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md [hunk 2]
```diff
@@ -204,3 +204,35 @@ for more information.
 ## Delta update support
 
 Delta update support is covered in [its own sub section](01.Delta-update-support/docs.md).
+
+
+## Addons
```
Comment (lluiscampos) [inline 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md:209]:
```suggestion
## Add-ons 
```

@merlin-northern right?

File: 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md [hunk 3]
```diff
@@ -204,3 +204,35 @@ for more information.
...
 
 Delta update support is covered in [its own sub section](01.Delta-update-support/docs.md).
+
+
+## Addons 
+
+### Mender-shell
+
+We do not enable Mender-shell by default, unless you are building with the demo layer.
+To enable Mender-shell you can add it either via your own `.bbappend` recipe file`,
```
Comment (lluiscampos) [inline 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md:214]:
```suggestion
To enable Mender-shell you can add it either via your own `.bbappend` recipe file,
```

File: 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md [hunk 4]
```diff
@@ -204,3 +204,35 @@ for more information.
...
+
+Mender-shell provides several [configuration options](../../03.Client-installation/06.Mender-shell-configuration-file/50.Mender-shell-configuration-options/docs.md). To override the default settings, create your own `mender-shell.conf` and
+augment the `mender-client` recipe with the new configuration. For example, create a `mender-client_%.bbappend` file in your layer, and add this:
+
+```bash
+FILESEXTRAPATHS_prepend := "${THISDIR}/<DIRECTORY-WITH-MENDER-SHELL-CONF>"
+SRC_URI_append = " file://mender-shell.conf"
+
+do_install_append() {
+    install -m 644 ${WORKDIR}/mender-shell.conf ${datadir}/mender/mender-shell.conf
```
Comment (lluiscampos) [inline 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md:233]: I think this should be `600` like in [the Makefile](https://github.com/mendersoftware/mender-shell/blob/master/Makefile#L59).

**Review by lluiscampos — CHANGES_REQUESTED** [review]
The PR should target the feature branch. See also some comments below.

The CI error is due to the broken link, which I think should be fix if we target feature branch.

File: 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md (hunk 2 above)
Comment (merlin-northern) [reply 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md:209]: yes, right!

File: 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md (hunk 4 above)
Comment (merlin-northern) [reply 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md:233]: corrected here https://github.com/mchalski/mender-docs/pull/1

File: 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md (hunk 3 above)
Comment (merlin-northern) [reply 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md:214]: corrected here https://github.com/mchalski/mender-docs/pull/1

File: 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md (hunk 2 above)
Comment (merlin-northern) [reply 05.System-updates-Yocto-Project/05.Customize-Mender/docs.md:209]: corrected here https://github.com/mchalski/mender-docs/pull/1

**Comment by merlin-northern** [comment]
needs a rebase on top of current feature branch, or we can merge it, the link is present, and I will be merging feature branch very soon any way.

**Comment by mchalski** [comment]
first off, thank you @merlin-northern for taking over this after discussions on doc structure.
I rebased on top of fresh `feature-deviceconnect` just now.

**Review by lluiscampos — APPROVED** [review]

**Pull request merged**